app/components/AddThis.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+
+interface Props {
+  url: string;
+  title: string;
+}
+
+declare const window: any;
+
+export default class Card extends React.Component<Props, {}> {
+  componentDidMount() {
+    window.addEventListener('load', window.addthis.layers.refresh());
+  }
+
+  componentDidUpdate() {
+    window.addEventListener('load', window.addthis.layers.refresh());
```

Comment:
Why are you adding new event listeners on every update? Didn't check, but this is almost certainly going to be a memory leak.

Reply:
You're right. Doesn't need it. This component doesn't get regular updates so wasn't a big deal, but you're right. Didn't need it. Really struggled with the lifecycle methods and where to put the refresh().